Allowing for dual output formats within one run

meatballhat opened the pull request:
so that a single run during provision can write both the
human-readable and json output formats.  Also added some minimal
help text.

meatballhat committed:
Allowing for dual output formats within one run

so that a single run during provision can write both the
human-readable and json output formats.  Also added some minimal
help text.

BanzaiMan commented on bin/system_info:
```diff
@@ -10,10 +29,9 @@ require 'timeout'
-COOKBOOKS_SHA = ARGV[0]
+ENV['FORMATS'] ||= 'human'
 
+COOKBOOKS_SHA = ENV['COOKBOOKS_SHA'] || ARGV[0]
```
Shouldn't the command line argument override the environment variable?

meatballhat replied:
Agreed.  Really,  I'm thinking this whole thing could at least use some optparse love, but that's for later :smiley:

meatballhat committed:
Swapping precedence of env var and command line arg

so that command line wins.

meatballhat committed:
Adding test to ensure 'Cookbooks Version' is written to human-readable format

meatballhat committed:
Passing the cookbooks version as command line arg

instead of env var, as the latter packs a lot of shell goop into an
env var definition, which is kinda mean.

meatballhat committed:
Passing correct mode arg to File.open

so that writing is possible.  Egad.

Commit:
Showing line after cookbooks version header as well

mostly for warm fuzzies

Commit:
Adding a json test and cat'ing everything at the end

for purposes of better reviewing the full terminal output.

Commit:
Installing jq if missing

as it hasn't landed yet as a base image tool.  yet.
